jpa/eclipselink.jpa.test.jse/src/org/eclipse/persistence/jpa/test/query/TestQueryCase.java

```diff
@@ -0,0 +1,191 @@
+ */
+// Contributors:
+//     IBM - Bug 537795: CASE THEN and ELSE scalar expression Constants should not be casted to CASE operand type
+package org.eclipse.persistence.jpa.test.query;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import java.util.List;
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.TypedQuery;
+import org.eclipse.persistence.jpa.test.framework.DDLGen;
+import org.eclipse.persistence.jpa.test.framework.Emf;
+import org.eclipse.persistence.jpa.test.framework.EmfRunner;
+import org.eclipse.persistence.jpa.test.framework.Property;
+import org.eclipse.persistence.jpa.test.query.model.Dto01;
+import org.eclipse.persistence.jpa.test.query.model.EntityTbl01;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(EmfRunner.class)
+public class TestQueryCase {
+    @Emf(createTables = DDLGen.DROP_CREATE, classes = { EntityTbl01.class }, 
+            properties = { @Property(name="eclipselink.logging.level", value="FINE") })
+    private EntityManagerFactory emf;
+
+    @BeforeClass
+    private void populate() {
+        EntityManager em = emf.createEntityManager();
+
+        try {
+            em.getTransaction().begin();
+
+            EntityTbl01 tbl1 = new EntityTbl01();
+            tbl1.setKeyString("Key01");
+            tbl1.setItemString1("047010");
+            tbl1.setItemString2("B");
+            tbl1.setItemString3("C");
+            tbl1.setItemString4("D");
+            tbl1.setItemInteger1(1000);
+            em.persist(tbl1);
+            EntityTbl01 tbl2 = new EntityTbl01();
+            tbl2.setKeyString("Key02");
+            tbl2.setItemString1("047010");
+            tbl2.setItemString2("B");
+            tbl2.setItemString3("C");
+            tbl2.setItemString4("D");
+            tbl2.setItemInteger1(100);
+            em.persist(tbl2);
+
+            em.getTransaction().commit();
+        } catch (Throwable t) {
+
+        }
+
+        em.close();
+    }
+
+    @Test
+    public void testQueryCase1() {
+        if (emf == null)
+            return;
+        EntityManager em = emf.createEntityManager();
+
+        em.getTransaction().begin();
+
+        TypedQuery<EntityTbl01> query = em.createQuery(""
+                + "SELECT t FROM EntityTbl01 t "
+                    + "WHERE t.itemString1 = ( "
+                        + "CASE t.itemInteger1 "
+                            + "WHEN 1000 THEN '047010' "
+                            + "WHEN 100 THEN '023010' "
+                            + "ELSE '033020' "
+                        + "END )", EntityTbl01.class);
+
+        List<EntityTbl01> dto01 = query.getResultList();
+        assertNotNull(dto01);
+        assertEquals(1, dto01.size());
+
+        em.getTransaction().rollback();
+    }
+
+    @Test
+    public void testQueryCase2() {
```

Comment:
pull in my most recent test updates (just committed)

Reply:
I have pulled in your test update